stdlib/contrib/bonitoo-io/tickscript/tickscript.flux

```diff
@@ -0,0 +1,156 @@
+package tickscript
+
+import "experimental"
+import "experimental/array"
+import "influxdata/influxdb"
+import "influxdata/influxdb/monitor"
+import "influxdata/influxdb/schema"
+import "universe"
+
+// defineCheck creates custom check data required by alert() and deadman()
+defineCheck = (id, name, type="custom") => {
+    return {
+        _check_id:   id,
+        _check_name: name,
+        _type:       type,
+        tags:        {}
+    }
+}
+
+// alert is a helper function similar to TICKscript alert.
+alert = (
+    check,
+    id=(r)=>"${r._check_id}",
+    details=(r)=>"",
+    message=(r)=>"Threshold Check: ${r._check_name} is: ${r._level}",
+    crit=(r) => false,
+    warn=(r) => false,
+    info=(r) => false,
+    ok=(r) => true,
+    topic="",
+    tables=<-) => {
+
+  _addTopic =
+    if topic != "" then
+      (tables=<-) => tables
+        |> set(key: "_topic", value: topic )
+        |> experimental.group(mode: "extend", columns: ["_topic"])
+    else
+      (tables=<-) => tables
+
+  return tables
+    |> drop(fn: (column) => column =~ /_start.*/ or column =~ /_stop.*/)
+    |> map(fn: (r) => ({r with
+        _check_id: check._check_id,
+        _check_name: check._check_name,
+    }))
+    |> map(fn: (r) => ({ r with id: id(r: r) }))
+    |> map(fn: (r) => ({ r with details: details(r: r) }))
+    |> _addTopic()
+    |> monitor.check(
+        crit: crit,
+        warn: warn,
+        info: info,
+        ok: ok,
+        messageFn: message,
+        data: check
+    )
+}
+
+// deadman is a helper function similar to TICKscript deadman.
+deadman = (
+    check,
+    measurement, threshold=0,
+    id=(r)=>"${r._check_id}",
+    message=(r)=>"Deadman Check: ${r._check_name} is: " + (if r.dead then "dead" else "alive"),
+    topic="",
+    tables=<-) => {
+
+   // In order to detect empty stream (without tables), it merges input with dummy stream and counts the result,
+   // because count() returns nothing for empty input. If the input stream is empty, then dummy stream with empty
+   // table is used in order for count() to return 0.
+
+  _dummy = array.from(rows: [{_time: 2000-01-01T00:00:00Z, _field: "unknown", _value: 0}])
+    |> set(key: "_measurement", value: measurement)
+    |> experimental.group(columns: ["_measurement"], mode: "extend") // required by monitor.check
+
+  _counts = union(tables: [_dummy, tables])
```

Comment:
I would expect this line to break as the records of `_dummy` and `tables` must be the same type and they are not in this case. 

If you use `aggregateWindow` with `createEmpty: true` it will create empty tables for the stream and then `count` will return `0` for those tables.

Reply:
I thought that `union()` just concatenates two streams and _if_ there are tables with the same schema they are concatenated in the resulting stream? It seems to work when I test it.

I cannot make stream with empty tables from bucket when I have no recent data, should I do it like this?
```
from(bucket: "jolokia")
  |> range(start: -10m)
  |> filter(fn: (r) => r._measurement == measurement, onEmpty: "keep")
  |> aggregateWindow(every: 5m, fn: count, createEmpty: true) 
```

I get totally empty stream.

Reply:
That query should work to get empty tables, you just have to ensure that the range is large enough to get at least some data for each series.

Reply:
Yes, with large enough range yes, but that may not be so common in the case of (deadman) alerts. Like in the set of TICKscripts from one customer, `deadman`s are set up with interval checks of only tens of minutes, typically 10m.